src/GNNGraphs/transform.jl

```diff
@@ -253,8 +254,33 @@ julia> g12.ndata.x
  1.0  1.0  1.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
 ```
 """
-Flux.batch(gs::Vector{<:GNNGraph}) = blockdiag(gs...)
-
+function Flux.batch(gs::Vector{<:GNNGraph})
+    nodes = [g.num_nodes for g in gs]
+
+    if all(y -> isa(y, COO_T), [g.graph for g in gs] )
+        edge_indices = [edge_index(g) for g in gs]
+        nodesum = cumsum([0, nodes...])[1:end-1]
+        s = reduce(vcat, [ei[1] .+ nodesum[ii] for (ii,ei) in enumerate(edge_indices)])
+        t = reduce(vcat, [ei[2] .+ nodesum[ii] for (ii,ei) in enumerate(edge_indices)])
+        w = reduce(vcat, [get_edge_weight(g) for g in gs])
+        w = w isa Vector{Nothing} ? nothing : w 
+        graph = (s, t, w)
+        graph_indicator = vcat([ones_like(ei[1],Int,nodes[ii]) .+ (ii - 1) for (ii,ei) in enumerate(edge_indices)]...)
+    elseif all(y -> isa(y, ADJMAT_T), [g.graph for g in gs] )
+        graph = blockdiag([g.graph for g in gs]...)
```

Comment:
this `blockdiag` method does not exist for dense matrices. 
Maybe we can use dispatch as follows:
```
Flux.batch(gs::Vector{<:GNNGraph{<:COO_T}) = ...  # specialized method
Flux.batch(gs::Vector{<:GNNGraph{<:SPARSE_T}) = ...  # specialized method
Flux.batch(gs::Vector{<:GNNGraph{<:SPARSE_T}) = blockdiag(gs...)  # old slow fallback

Reply:
Thanks for the comments! I'll work on the dispatching